File: gpp-api/src/main/java/com/develop/gpp/domain/entity/DepartamentoModel.java
package com.develop.gpp.domain.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "departamento")
public class DepartamentoModel {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id_departamento")
  private long idDepartamento;

  @Column(name = "nome")
  private String nome;

  @Column(name = "situacao")
  private boolean situacao;

  @Column(name = "created_at")
  private LocalDateTime createdAt;

  @Column(name = "updated_at")
  private LocalDateTime updatedAt;

  // Constructors, getters, and setters omitted for brevity
}

File: angular-gpp/src/app/models/DepartamentoModel.ts
export class DepartamentoModel {
  idDepartamento: number;
  nome: string;
  situacao: boolean;
  createdAt: string;
  updatedAt: string;

  constructor(idDepartamento: number, nome: string, situacao: boolean, createdAt: string, updatedAt: string) {
    this.idDepartamento = idDepartamento;
    this.nome = nome;
    this.situacao = situacao;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
  }
}
